libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1594,11 +1594,18 @@ bool TypeChecker::visit(UnaryOperation const& _operation)
 		requireLValue(_operation.subExpression(), false);
 	else
 		_operation.subExpression().accept(*this);
+
 	Type const* subExprType = type(_operation.subExpression());
-	Type const* t = type(_operation.subExpression())->unaryOperatorResult(op);
-	if (!t)
+	TypeResult result = subExprType->unaryOperatorResult(op);
+	Type const* t = result;
```

Comment:
Can't you just remove this and replace every ``t`` below with ``result``? If not: also fine.

Reply:
I remember I actually tried that and that there was a problem with that. Something with `TypeResult` vs `Type`..

Reply:
Ah yes. We're assigning to `t` in line 1615 and that's why. Assigning to a `TypeResult` is at the very least confusing here

Reply:
In my mind assigning to ``TypeResult`` is not that big of a deal (although not sure if it just works and jumping through hoops for being able to would definitely be too much). I'm fine with keeping the ``t``.